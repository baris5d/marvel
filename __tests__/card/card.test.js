import Card from "../../components/card";
import {shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson  } from 'enzyme-to-json';

import renderer from 'react-test-renderer';

configure({adapter: new Adapter()});

const initialProps = {
  src: "test.jpg",
  name: "test",
  id: "0"
};

const wrapper = shallow(<Card />);

describe("Card Component", () => {
  afterAll(() => wrapper.unmount());

  it("shouldn't render a card component without props", () => {
    expect(wrapper.type()).toBeNull();
  });

  it("should render a card if props present", () => {
    wrapper.setProps({ ...initialProps });

    expect(wrapper.find(".characterImage")).toHaveLength(1);
    expect(wrapper.find("Image").prop("src")).toBe(initialProps.src);
    expect(wrapper.find("Image").prop("alt")).toBe(initialProps.name);
    expect(wrapper.find("h2").prop("children")).toBe(initialProps.name);
  });

  it("renders correctly", () => {
        wrapper.setProps({ ...initialProps });
        wrapper.update()
        expect(shallowToJson(wrapper)).toMatchSnapshot();
  })
});