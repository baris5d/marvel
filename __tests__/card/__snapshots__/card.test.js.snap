// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders correctly 1`] = `
<div
  className="card"
  onClick={[Function]}
>
  <Image
    alt="test"
    className="characterImage"
    src="test.jpg"
  />
  <h2
    className="characterName"
  >
    test
  </h2>
</div>
`;
